### PR TITLE
Catch NULL from malloc()

### PR DESCRIPTION
**Pull request by KlzXS**

### CODE REVIEW

File: src/nnn.c [hunk 1]
```diff
@@ -2628,8 +2633,13 @@ static char *get_kv_val(kv *kvarr, char *buf, int key, uchar max, bool path)
 				ssize_t len = strlen(home);
 				ssize_t loclen = strlen(kvarr[r].val);
 
-				if (!buf)
+				if (!buf) {
 					buf = (char *)malloc(len + loclen);
+					if (!buf) {
+						DPRINTF_S("malloc failed!");
```
Comment (jarun) [inline src/nnn.c]: Please log `strerror(errno)`.

File: src/nnn.c [hunk 2]
```diff
@@ -1676,6 +1676,11 @@ static void get_archive_cmd(char *cmd, char *archive)
 static void archive_selection(const char *cmd, const char *archive, const char *curpath)
 {
 	char *buf = (char *)malloc(CMD_LEN_MAX * sizeof(char));
+	if (!buf) {
```
Comment (jarun) [inline src/nnn.c]: If we look at what we are trying to store in buf, we need to allocate:

strlen(cmd) + strlen(archive) + strlen(curpath) + strlen(g_selpath);

File: src/nnn.c [hunk 3]
```diff
@@ -1676,6 +1676,11 @@ static void get_archive_cmd(char *cmd, char *archive)
...
+		DPRINTF_S("malloc failed!");
+		printmsg(messages[MSG_FAILED]);
```
Comment (jarun) [inline src/nnn.c]: please use `printwarn(NULL)`;

File: src/nnn.c [hunk 4]
```diff
@@ -1676,6 +1676,11 @@ static void get_archive_cmd(char *cmd, char *archive)
 static void archive_selection(const char *cmd, const char *archive, const char *curpath)
 {
 	char *buf = (char *)malloc(CMD_LEN_MAX * sizeof(char));
+	if (!buf) {
+		DPRINTF_S("malloc failed!");
```
Comment (jarun) [inline src/nnn.c]: please log `strerror(errno)`.

File: src/nnn.c (hunk 2 above)
Comment (KlzXS) [reply src/nnn.c]: You want me so allocate this amount instead?

Comment (jarun) [reply src/nnn.c]: Yes,` CMD_LEN_MAX` is insufficient.

Comment (KlzXS) [reply src/nnn.c]: You forgot to count the length of the string we snprintf() into, that's an aditional ~70.

**Comment by jarun** [comment]
Thank you!

**Pull request merged**